src/ucp/stream/stream.h

```diff
@@ -15,6 +15,15 @@ typedef struct {
 } UCS_S_PACKED ucp_stream_am_hdr_t;
 
 
+typedef struct {
+    union {
+        ucp_stream_am_hdr_t hdr;
+        size_t              offset;
+    };
+    uint8_t                 payload[0];
```

Comment:
i would remove this field and use pointer arithmetic to reach the payload

Reply:
When we will have 2 different assessors to `offset` and the payload...